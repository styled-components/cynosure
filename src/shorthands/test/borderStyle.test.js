// @flow
import borderStyle from '../borderStyle'

describe('borderStyle', () => {
  it('properly applies a value when passed only one', () => {
    expect(borderStyle('solid')).toEqual({
      borderBottomStyle: 'solid',
      borderLeftStyle: 'solid',
      borderRightStyle: 'solid',
      borderTopStyle: 'solid',
    })
  })
  it('properly applies values when passed two', () => {
    expect(borderStyle('solid', 'dashed')).toEqual({
      borderBottomStyle: 'solid',
      borderLeftStyle: 'dashed',
      borderRightStyle: 'dashed',
      borderTopStyle: 'solid',
    })
  })
  it('properly applies values when passed three', () => {
    expect(borderStyle('solid', 'dashed', 'dotted')).toEqual({
      borderBottomStyle: 'dotted',
      borderLeftStyle: 'dashed',
      borderRightStyle: 'dashed',
      borderTopStyle: 'solid',
    })
  })
  it('properly applies values when passed four', () => {
    expect(borderStyle('solid', 'dashed', 'dotted', 'double')).toEqual({
      borderBottomStyle: 'dotted',
      borderLeftStyle: 'double',
      borderRightStyle: 'dashed',
      borderTopStyle: 'solid',
    })
  })
})
