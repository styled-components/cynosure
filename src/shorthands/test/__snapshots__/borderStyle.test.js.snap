exports[`borderStyle properly applies a value when passed only one 1`] = `
Object {
  "borderBottomStyle": "solid",
  "borderLeftStyle": "solid",
  "borderRightStyle": "solid",
  "borderTopStyle": "solid",
}
`;

exports[`borderStyle properly applies values when passed four 1`] = `
Object {
  "borderBottomStyle": "dotted",
  "borderLeftStyle": "double",
  "borderRightStyle": "dashed",
  "borderTopStyle": "solid",
}
`;

exports[`borderStyle properly applies values when passed three 1`] = `
Object {
  "borderBottomStyle": "dotted",
  "borderLeftStyle": "dashed",
  "borderRightStyle": "dashed",
  "borderTopStyle": "solid",
}
`;

exports[`borderStyle properly applies values when passed two 1`] = `
Object {
  "borderBottomStyle": "solid",
  "borderLeftStyle": "dashed",
  "borderRightStyle": "dashed",
  "borderTopStyle": "solid",
}
`;
